blockchain/fullblocktests/generate.go

```diff
@@ -2149,6 +2160,43 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 	})
 	rejected(blockchain.ErrImmatureSpend)
 
+	// Create block with a malformed outputs order for a ticket purchase.
+	//
+	//   ... -> brs3(14)
+	//                  \-> bmf36(15)
+	g.SetTip("brs3")
+	g.NextBlock("bmf36", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		prevBlock := g.Tip()
+		spendOut := chaingen.MakeSpendableOut(prevBlock, 1, 0)
+		ticketPrice := dcrutil.Amount(g.CalcNextRequiredStakeDifficulty())
+		ticket := g.CreateTicketPurchaseTx(&spendOut, ticketPrice, lowFee)
+		tmp := ticket.TxOut[0]
+		ticket.TxOut[0] = ticket.TxOut[2]
+		ticket.TxOut[2] = tmp
+		b.AddSTransaction(ticket)
+		b.Header.FreshStake++
+	})
+	rejected(blockchain.ErrRegTxInStakeTree)
+
+	// Create block with scripts that do not involve p2pkh or
+	// p2sh addresses for a ticket purchase.
+	//
+	//   ... -> brs3(14)
+	//                  \-> bmf37(15)
+	g.SetTip("brs3")
+	g.NextBlock("bmf37", outs[15], ticketOuts[15], func(b *wire.MsgBlock) {
+		prevBlock := g.Tip()
+		spendOut := chaingen.MakeSpendableOut(prevBlock, 1, 0)
+		ticketPrice := dcrutil.Amount(g.CalcNextRequiredStakeDifficulty())
+		ticket := g.CreateTicketPurchaseTx(&spendOut, ticketPrice, lowFee)
+		ticket.TxOut[0].PkScript = opTrueScript
+		ticket.TxOut[1].PkScript = opTrueScript
```

Comment:
I would suggest limiting it to output 0 since that is the one that confers voting rights.  Also, a block in the tests should only test a single failure at a time to be sure the intended failure is actually the one being tested.